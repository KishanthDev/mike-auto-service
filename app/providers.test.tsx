import React from 'react';
import { render } from '@testing-library/react';
import { Providers } from './providers';
import { HeroUIProvider } from '@heroui/system';
import { ThemeProvider as NextThemesProvider } from 'next-themes';
import { useRouter } from 'next/navigation';

// Mock the dependencies
jest.mock('@heroui/system', () => ({
  HeroUIProvider: jest.fn(({ children }) => <div>{children}</div>),
}));

jest.mock('next-themes', () => ({
  ThemeProvider: jest.fn(({ children }) => <div>{children}</div>),
}));

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(() => ({
    push: jest.fn(),
  })),
}));

describe('Providers Component', () => {
  const mockChildren = <div data-testid="test-child">Test Content</div>;
  const mockThemeProps = {
    attribute: 'class',
    defaultTheme: 'dark',
  };

  beforeEach(() => {
    // Clear all mocks before each test
    jest.clearAllMocks();
  });


  it('wraps children with HeroUIProvider and NextThemesProvider', () => {
    render(<Providers>{mockChildren}</Providers>);

    expect(HeroUIProvider).toHaveBeenCalled();
    expect(NextThemesProvider).toHaveBeenCalled();

    // Verify the nesting order
    const heroUIProviderCall = (HeroUIProvider as jest.Mock).mock.calls[0][0];
    expect(heroUIProviderCall.children.type).toBe(NextThemesProvider);
  });

  it('passes router.push to HeroUIProvider navigate prop', () => {
    const mockPush = jest.fn();
    (useRouter as jest.Mock).mockReturnValueOnce({ push: mockPush });

    render(<Providers>{mockChildren}</Providers>);

    expect(HeroUIProvider).toHaveBeenCalledWith(
      expect.objectContaining({
        navigate: mockPush,
      }),
      expect.anything()
    );
  });

  it('uses default theme props when none are provided', () => {
    render(<Providers>{mockChildren}</Providers>);

    expect(NextThemesProvider).toHaveBeenCalledWith(
      expect.objectContaining({}),
      expect.anything()
    );
  });

  it('passes children through to NextThemesProvider', () => {
    render(<Providers>{mockChildren}</Providers>);

    const lastThemesProviderCall = (NextThemesProvider as jest.Mock).mock.calls[0][0];
    expect(lastThemesProviderCall.children).toBe(mockChildren);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Providers>{mockChildren}</Providers>);
    expect(asFragment()).toMatchSnapshot();
  });
});