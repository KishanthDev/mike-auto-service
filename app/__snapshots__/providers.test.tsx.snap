// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Providers Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        data-testid="test-child"
      >
        Test Content
      </div>
    </div>
  </div>
</DocumentFragment>
`;
